plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/KubernetesClusterManagerImpl.java

```diff
@@ -1384,7 +1384,15 @@ private String[] getServiceUserKeys(KubernetesClusterVO kubernetesCluster) {
         if (owner == null || owner.getType() == Account.Type.PROJECT) {
             owner = CallContext.current().getCallingAccount();
         }
-        String username = owner.getAccountName() + "-" + KUBEADMIN_ACCOUNT_NAME;
+        
+        String projectId = "default";
+        Account account = ApiDBUtils.findAccountById(kubernetesCluster.getAccountId());
+        Project project = ApiDBUtils.findProjectByProjectAccountId(account.getId());
+        if ( project != null ) {
+          projectId = project.getUuid();
+        }
+
+        String username = owner.getAccountName() + "-" + projectId + "-" + KUBEADMIN_ACCOUNT_NAME;
```

Comment:
does this mean accounts cannot have a kubernetes instance without being part of a project? I would expect something like 
```suggestion
        String projectId = null;
        Account account = ApiDBUtils.findAccountById(kubernetesCluster.getAccountId());
        Project project = ApiDBUtils.findProjectByProjectAccountId(account.getId());
        if ( project != null ) {
          projectId = project.getUuid();
        }

        String username = owner.getAccountName() + (projectId == null ? : "" : "-" + projectId) + "-" + KUBEADMIN_ACCOUNT_NAME;
```

Reply:
That's a good idea. Will update the code as such. In our case, we always have projectId.